### DISCUSSION TITLE
Next.js middleware on Vercel not able to query SB public schema?

### DISCUSSION THREAD

**Discussion by riccardolardi** [supabase/supabase#20245] [Questions] [answered]
I'm trying to query Supabase public schema "profiles" table on a Next.js project in middleware while being hosted on Vercel:

```ts
import { createMiddlewareClient } from "@supabase/auth-helpers-nextjs";
import { NextResponse } from "next/server";
import type { NextRequest } from "next/server";

export async function middleware(req: NextRequest) {
  const res = NextResponse.next();
  const supabase = createMiddlewareClient({ req, res });

  const {
    data: { session },
  } = await supabase.auth.getSession();

  const data = await supabase
    .from("profiles")
    .select("role")
    .eq("id", session?.user.id);
  console.log({ data });

  return res;
}

```

But I'm getting this error:
```js
{
  data: {
    error: {
    code: '42501',
    details: null,
    hint: null,
    message: 'permission denied for schema public'
  },
    data: null,
    count: null,
    status: 403,
    statusText: 'Forbidden'
  }
}
```

Is that a bug in Supabase or why is it not possible to query?

**Comment by GaryAustin1** [comment]
It is unlikely a Supabase bug.
Do you use Prisma?  Have you modified the public schema user grants?

**Reply by riccardolardi** [reply]
Thanks @GaryAustin1 yes we do use Prisma but it should have no implication in this case, should it? I'm not using here in middleware, instead we're using the Supabase JS SDK directly...

When run locally it works fine. That's why I was thinking it might have to do with Vercel Edge functions. Isn't that possible?

**Reply by GaryAustin1** [reply, marked as answer]
Yes Prisma migrations can wipe the grants on the public schema so Supabase roles have no access.  See troubleshooting here https://supabase.com/partners/integrations/prisma .

**Reply by riccardolardi** [reply]
Thank you @GaryAustin1 that did the trick - the grants had been wiped by prisma.
